### DISCUSSION TITLE
Computer not sleeping

### DISCUSSION THREAD

**Discussion by samchouse** [arkane-systems/bottle-imp#20] [Q&A] [answered]
Does Imp prevent the computer from sleeping with it's `Keeps the WSL instance running even when you have no active sessions.` feature? I recently started using Imp and my PC is falling asleep anymore.

**Comment by cerebrate** [comment, marked as answer]
That is a somewhat difficult question to answer.

_imp_ itself does not intentionally do anything to keep the computer awake. (In fact, if anything, it goes out of its way to minimize the potential impact it has on making the system appear non-idle; the script that does the job sleeps for an hour at a time to avoid creating any CPU load. And obviously, as a WSL process, it can't call _SetThreadExecutionState()_ to mark its thread as no-display-sleep or no-system-sleep even if it wanted to.)

However, that said, Windows itself determines when to sleep by looking for "user or application activity". The details of this are obscure, but it does include both user input and, more relevantly, network communications. So, with _systemd_ services running in the background, and not subject to automatic termination when you close all your WSL sessions, it's possible that those services are generating enough aggregate network communication activity (or other unspecified criteria) to make Windows think that the system is non-idle and thus stop it from going into sleep state.

That's what I suspect is most likely, anyway.

**Reply by samchouse** [reply]
I did some further digging, this was caused by an application that was installed around the same time as Imp. Thanks for the detailed explanation though :)
